server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -83,7 +83,12 @@ public Map<String, Long> toBreakdownMap() {
             // creates a new weight and breakdown map for each rewritten query. This new breakdown map captures the timing information for
             // the new rewritten query. The sliceCollectorsToLeaves is empty because this breakdown for rewritten query gets created later
             // in search leaf path which doesn't have collector. Also, this is not needed since this breakdown is per leaf and there is no
-            // concurrency involved. An empty sliceCollectorsToLeaves could also happen in the case of early termination.
+            // concurrency involved.
+            assert contexts.size() == 1 : new OpenSearchException(
```

Comment:
Why do we need to create `OpenSearchException` here? 

```suggestion
            assert contexts.size() == 1 : "Unexpected size: "
                    + contexts.size() + " of leaves breakdown in ConcurrentQueryProfileBreakdown of rewritten query for a leaf."
```

should be enough?

Reply:
Hi @reta, thanks for the review. This assertion could help us to catch unknown edge cases. For example, before this PR, post filter profile breakdown could have more than 1 context here. And `assert contexts.size() == 1` is because we only expect 1 context in the case mentioned from L82 to L86.

Reply:
Thanks @ticheng-aws , the assertion is fine, I was wondering why we create `OpenSearchException` here?

Reply:
Got it... thanks. It has been addressed now.